fvm/Equation/MomentQuantity.cpp

```diff
@@ -66,10 +204,38 @@ bool MomentQuantity::GridRebuilt() {
  * x:       Value of the unknown quantity.
  */
 void MomentQuantity::SetJacobianBlock(
-    const len_t unknId, const len_t derivId, Matrix *jac, const real_t* /*x*/
+    const len_t unknId, const len_t derivId, Matrix *jac, const real_t* x
 ) {
-    if (derivId == fId && unknId == fId)
+    if (derivId == unknId)
```

Comment:
this should be ok right? unknId should in practice always be fId

Reply:
For a given ``MomentQuantity`` *object*, ``unknId`` will only take one value, and that is the ID of the unknown to which the operator is applied. So as long as all ``MomentQuantity``s are applied to ``f``, this should be fine.

Reply:
(actually, it's probably always fine. I think this is exactly the intended behaviour in every case where the ``MomentQuantity`` is involved)